adapters/consumable/consumable.go

```diff
@@ -3,319 +3,154 @@ package consumable
+			bidType, err := getMediaTypeForBid(bid)
+			if err != nil {
+				errors = append(errors, err)
+				continue
 			}
-		}
-	}
-
-	if request.User != nil && request.User.Ext != nil {
-		var extUser openrtb_ext.ExtUser
-		if err := json.Unmarshal(request.User.Ext, &extUser); err != nil {
-			errs = append(errs, err)
-		} else {
-			gdpr.Consent = extUser.Consent
-			body.GDPR = &gdpr
-
-			if hasEids(extUser.Eids) {
-				body.User.Eids = extUser.Eids
+			var bidVideo *openrtb_ext.ExtBidPrebidVideo
+			if bidType == openrtb_ext.BidTypeVideo {
+				bidVideo = &openrtb_ext.ExtBidPrebidVideo{Duration: int(bid.Dur)}
+			}
+			switch bidType {
+			case openrtb_ext.BidTypeAudio:
+				seatBid.Bid[i].MType = openrtb2.MarkupAudio
+				break
+			case openrtb_ext.BidTypeVideo:
+				seatBid.Bid[i].MType = openrtb2.MarkupVideo
+				break
+			case openrtb_ext.BidTypeBanner:
+				seatBid.Bid[i].MType = openrtb2.MarkupBanner
+				break
 			}
+			bidResponse.Bids = append(bidResponse.Bids, &adapters.TypedBid{
+				Bid:      &seatBid.Bid[i],
+				BidType:  bidType,
+				BidVideo: bidVideo,
+			})
 		}
 	}
+	return bidResponse, nil
+}
 
-	if request.Source != nil && request.Source.Ext != nil {
-		var extSChain openrtb_ext.ExtRequestPrebidSChain
-		if err := json.Unmarshal(request.Source.Ext, &extSChain); err != nil {
-			errs = append(errs, err)
-		} else {
-			body.SChain = extSChain.SChain
+func getMediaTypeForBid(bid openrtb2.Bid) (openrtb_ext.BidType, error) {
+	if bid.MType != 0 {
+		switch bid.MType {
+		case openrtb2.MarkupBanner:
+			return openrtb_ext.BidTypeBanner, nil
+		case openrtb2.MarkupVideo:
+			return openrtb_ext.BidTypeVideo, nil
+		case openrtb2.MarkupAudio:
+			return openrtb_ext.BidTypeAudio, nil
 		}
 	}
-
-	body.Coppa = request.Regs != nil && request.Regs.COPPA > 0
-
-	if request.Regs != nil && request.Regs.GPP != "" {
-		body.GPP = request.Regs.GPP
-	}
-
-	if request.Regs != nil && request.Regs.GPPSID != nil {
-		body.GPPSID = request.Regs.GPPSID
-	}
-
-	if request.Site != nil && request.Site.Content != nil {
-		body.Content = request.Site.Content
-	} else if request.App != nil && request.App.Content != nil {
-		body.Content = request.App.Content
-	}
-
-	for i, impression := range request.Imp {
-
-		_, consumableExt, err := extractExtensions(impression)
-
-		if err != nil {
-			return nil, err
-		}
-
-		// These get set on the first one in observed working requests
-		if i == 0 {
-			body.NetworkId = consumableExt.NetworkId
-			body.SiteId = consumableExt.SiteId
-			body.UnitId = consumableExt.UnitId
-			body.UnitName = consumableExt.UnitName
-		}
-
-		body.Placements[i] = placement{
-			DivName:   impression.ID,
-			NetworkId: consumableExt.NetworkId,
-			SiteId:    consumableExt.SiteId,
-			UnitId:    consumableExt.UnitId,
-			UnitName:  consumableExt.UnitName,
-			AdTypes:   getSizeCodes(impression.Banner.Format), // was adTypes: bid.adTypes || getSize(bid.sizes) in prebid.js
+	if bid.Ext != nil {
+		var bidExt openrtb_ext.ExtBid
```

Comment:
`bid.MType` is used on line 125. Could we remove `bid.Ext` check and rely on `bid.MType`

Reply:
To be honest the only reason the bid.Ext stuff is in there is because that is what is on the documentation site as the example